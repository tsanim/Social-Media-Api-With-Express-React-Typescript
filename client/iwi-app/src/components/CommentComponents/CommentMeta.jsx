import React from 'react';
import calcTime from '../../utils/calcTime';
import PropTypes from 'prop-types';

function CommentMeta(props) {
    return (
        <div className="meta">
            <span className="date">{calcTime(props.date)}</span>
            <span onClick={props.handleShowLikesModal} >{props.likesCount} likes</span>
        </div>
    )
}

CommentMeta.propTypes = {
    date: PropTypes.string,
    likesCount: PropTypes.number,
    handleShowLikesModal: PropTypes.func,
    testProp: PropTypes.string
}

export default CommentMeta;