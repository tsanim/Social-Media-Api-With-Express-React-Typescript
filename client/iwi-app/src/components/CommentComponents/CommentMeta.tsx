import React from 'react';
import calcTime from '../../utils/calcTime';
import PropTypes from 'prop-types';
import CommentMetaProps from '../../interfaces/Components/CommentComponents/CommentMetaProps.interface';

function CommentMeta(props: CommentMetaProps) {
    return (
        <div className="meta">
            <span className="date">{calcTime(props.date)}</span>
            <span onClick={props.handleShowLikesModal} >{props.likesCount} likes</span>
        </div>
    )
}

CommentMeta.propTypes = {
    date: PropTypes.string,
    likesCount: PropTypes.number,
    handleShowLikesModal: PropTypes.func,
}

export default CommentMeta;