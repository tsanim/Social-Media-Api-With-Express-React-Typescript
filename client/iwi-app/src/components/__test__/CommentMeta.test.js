import React from 'react';
import CommentMeta from '../CommentComponents/commentMeta';

import { shallow } from 'enzyme';

function commentMetaComponentSetup() {
    const props = {
        date: '',
        likesCount: 1,
        handleShowLikesModal: jest.fn(),
    }

    const enzymeWrapper = shallow(<CommentMeta {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('CommentMeta component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = commentMetaComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div with class "meta"', () => {
        const { enzymeWrapper } = commentMetaComponentSetup();

        expect(enzymeWrapper.find('div.meta')).toHaveLength(1);
    });

    it('should render two span tags', () => {
        const { enzymeWrapper } = commentMetaComponentSetup();

        expect(enzymeWrapper.find('span')).toHaveLength(2);
    });

    it('should render span tag with class "date"', () => {
        const { enzymeWrapper } = commentMetaComponentSetup();

        expect(enzymeWrapper.find('span.date')).toHaveLength(1);
    });
});
