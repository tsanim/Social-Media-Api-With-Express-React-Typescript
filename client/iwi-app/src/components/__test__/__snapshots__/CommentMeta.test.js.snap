// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentMeta component should render initial layout 1`] = `
Array [
  <div
    className="meta"
  >
    <span
      className="date"
    >
      NaN years
    </span>
    <span
      onClick={[MockFunction]}
    >
      1
       likes
    </span>
  </div>,
]
`;
